speech/google/cloud/speech/transcript.py

```diff
@@ -38,7 +38,7 @@ def from_api_repr(cls, transcript):
         :rtype: :class:`Transcript`
         :returns: Instance of ``Transcript``.
         """
-        return cls(transcript['transcript'], transcript['confidence'])
+        return cls(transcript.get('transcript'), transcript.get('confidence'))
```

Comment:
I put the indexing back since that doesn't really have anything to do with the issue.
https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2646/commits/f7fb81895d6f433b6561f909fb7a695d78bd42cc